lib/segment/src/index/query_optimization/condition_converter/match_converter.rs

```diff
@@ -9,38 +10,47 @@ use crate::types::{
     AnyVariants, Match, MatchAny, MatchExcept, MatchText, MatchValue, ValueVariants,
 };
 
-pub fn get_match_checkers(index: &FieldIndex, cond_match: Match) -> Option<ConditionCheckerFn> {
+pub fn get_match_checkers(
+    index: &FieldIndex,
+    cond_match: Match,
+    hw_acc: HwMeasurementAcc,
+) -> Option<ConditionCheckerFn> {
     match cond_match {
-        Match::Value(MatchValue { value }) => get_match_value_checker(value, index),
-        Match::Text(MatchText { text }) => get_match_text_checker(text, index),
-        Match::Any(MatchAny { any }) => get_match_any_checker(any, index),
-        Match::Except(MatchExcept { except }) => get_match_except_checker(except, index),
+        Match::Value(MatchValue { value }) => get_match_value_checker(value, index, hw_acc),
+        Match::Text(MatchText { text }) => get_match_text_checker(text, index, hw_acc),
+        Match::Any(MatchAny { any }) => get_match_any_checker(any, index, hw_acc),
+        Match::Except(MatchExcept { except }) => get_match_except_checker(except, index, hw_acc),
     }
 }
 
 fn get_match_value_checker(
     value_variant: ValueVariants,
     index: &FieldIndex,
+    hw_acc: HwMeasurementAcc,
 ) -> Option<ConditionCheckerFn> {
     match (value_variant, index) {
         (ValueVariants::String(keyword), FieldIndex::KeywordIndex(index)) => {
+            let hw_counter = hw_acc.get_counter_cell();
             Some(Box::new(move |point_id: PointOffsetType| {
-                index.check_values_any(point_id, |k| k == keyword)
+                index.check_values_any(point_id, &hw_counter, |k| k == keyword)
             }))
         }
         (ValueVariants::String(value), FieldIndex::UuidMapIndex(index)) => {
             let uuid = Uuid::parse_str(&value).map(|uuid| uuid.as_u128()).ok()?;
+            let hw_counter = hw_acc.get_counter_cell();
             Some(Box::new(move |point_id: PointOffsetType| {
-                index.check_values_any(point_id, |i| i == &uuid)
+                index.check_values_any(point_id, &hw_counter, |i| i == &uuid)
             }))
         }
         (ValueVariants::Integer(value), FieldIndex::IntMapIndex(index)) => {
+            let hw_counter = hw_acc.get_counter_cell();
             Some(Box::new(move |point_id: PointOffsetType| {
-                index.check_values_any(point_id, |i| *i == value)
+                index.check_values_any(point_id, &hw_counter, |i| *i == value)
             }))
         }
         (ValueVariants::Bool(is_true), FieldIndex::BoolIndex(index)) => {
             Some(Box::new(move |point_id: PointOffsetType| {
+                // TODO(io_measurement): implement measurement
                 index.check_values_any(point_id, is_true)
```

Comment:
Can't we count just 1 for a boolean check?

In the description I see that you're handling this in a separate PR, fine by me. :+1:

Reply:
In theory it is more than 1 byte though because we load a full page into memory, even if we only check one byte.
Can we maybe go probabilistic here and say 99% of the time we access the same page so we use
`0.01 *  PAGE_SIZE` (4kb) as constant measurement.
@generall wdyt?

Reply:
Looks like it's already implemented with 1 byte for boolean here: https://github.com/qdrant/qdrant/pull/5954/files#diff-2bee7d830875853065cb32b763bc13bc85fca02f0dbab25f7018d251fd192d20R186

I think it's still worth reconsidering to use > 1 byte.